libtenzir/src/node.cpp

```diff
@@ -330,6 +383,23 @@ node(node_actor::stateful_pointer<node_state> self, std::string /*name*/,
       }
     }
   });
+  self->set_exception_handler([=](std::exception_ptr& ptr) -> caf::error {
+    caf::error err;
+    try {
+      std::rethrow_exception(ptr);
+    } catch (const diagnostic& diag) {
+      err = diag.to_error();
+      TENZIR_ERROR("{}", err);
+    } catch (...) {
+      auto x = std::current_exception();
+      err = caf::scheduled_actor::default_exception_handler(self, x);
+      if (err) {
+        self->call_error_handler(err);
+      }
+    }
+    self->call_error_handler(err);
```

Comment:
This could call the error handler twice, no?

Reply:
I'm not even sure why we need to call this. We don't in the other custom `exception_handler`s and neither do the CAF unit tests.